api/src/controllers/files.ts

```diff
@@ -92,28 +104,47 @@ export const multipartHandler: RequestHandler = (req, res, next) => {
 		try {
 			const primaryKey = await service.uploadOne(fileStream, payloadWithRequiredFields, existingPrimaryKey);
 			savedFiles.push(primaryKey);
-			tryDone();
 		} catch (error: any) {
-			busboy.emit('error', error);
+			errors.push(error);
+		} finally {
+			totalProcessedFiles++;
+			tryDone();
 		}
 	});
 
-	busboy.on('error', (error: Error) => {
+	
+
+	// On Busboy file count limit error
+	busboy.on('filesLimit', () => {
+		errors.push(
+			new MaxFileCountExceededException(
+				`The maximum number of files allowed to be uploaded at once is ${env.ASSETS_LIMIT_FILE_COUNT}`
+			)
+		);
+		tryDone();
+	});
+
+	busboy.on('error', (error: any) => {
 		next(error);
 	});
 
 	busboy.on('close', () => {
 		tryDone();
 	});
 
+	busboy.on('finish', () => {
+		tryDone();
+	});
+
 	req.pipe(busboy);
 
 	function tryDone() {
-		if (savedFiles.length === fileCount) {
+		if (totalProcessedFiles === fileCount || totalProcessedFiles === limits.files) {
```

Comment:
This will end processing files at the limit OR when it has finished uploading/erroring all files in the form data.

Which means that the File Count Limit is INCLUSIVE of errors. So if the limit is 2, and file #2 errors, file#3 will not be processed because it stopped after 2 files came in.